apm-commons/apm-datacarrier/src/main/java/org/apache/skywalking/apm/commons/datacarrier/common/AtomicRangeInteger.java

```diff
@@ -18,54 +18,106 @@
 
 
 package org.apache.skywalking.apm.commons.datacarrier.common;
+import sun.misc.Unsafe;
```

Comment:
Is this safe for all JREs? Including OpenJDK and new versions? I remember Oracle changed the package name once.

Reply:
Yes,  Unsafe maybe change in the future, I have changed the code, just  a small change on the v2. 
This implementation also used the L1 cache, so it's very fast.
Disable the false-sharing reference from the Disruptor, Disruptor's Sequence also use placeholders to full a cache line so that the increase value changes doesn't affect others cache.

see [Sequence](https://github.com/LMAX-Exchange/disruptor/blob/master/src/main/java/com/lmax/disruptor/Sequence.java)